src/epd2in13bc/graphics.rs

```diff
@@ -1,38 +1,41 @@
+use crate::color::TriColor;
 use crate::epd2in13bc::{DEFAULT_BACKGROUND_COLOR, HEIGHT, NUM_DISPLAY_BITS, WIDTH};
-use crate::graphics::{Display, DisplayRotation};
-use embedded_graphics::pixelcolor::BinaryColor;
+use crate::graphics::{DisplayRotation, TriDisplay};
 use embedded_graphics::prelude::*;
 
-/// Full size buffer for use with the 2in13b/c EPD
+/// Full size buffer for use with the 2.13" b/c EPD
 ///
 /// Can also be manually constructed and be used together with VarDisplay
 pub struct Display2in13bc {
-    buffer: [u8; NUM_DISPLAY_BITS as usize],
+    // one buffer for both b/w and for chromatic:
+    // * &buffer[0..NUM_DISPLAY_BITS] for b/w buffer and
+    // * &buffer[NUM_DISPLAY_BITS..2*NUM_DISPLAY_BITS] for chromatic buffer
+    buffer: [u8; 2 * NUM_DISPLAY_BITS as usize],
```

Comment:
I am not sure if this is really a good change, since it doubles the space requirements :thinking:

Reply:
To write to the epd, the data is sent using [interface::write](https://github.com/caemor/epd-waveshare/blob/401858c4565343e9ec561cd9bfa61da3bc32c542/src/interface.rs#L99) which takes a slice of the buffer. For tri-color a slice for the b/w-buffer or a slice to the white/chromatic buffer. This means that whatever datatype you choose here, I think you always need to convert it into those two buffers when updating the display probably allocating at each refresh

So I'm not sure how to avoid doubling space requirements.

Reply:
But aren't they written after each other? So there is still potentially the option to just replace the content of one of the buffers before sending it to the device. If that is not the case, we can just go forward with your change. And even if its not I think your change should still be fine, since a user with special needs should be able to use the VarDisplay instead :thinking:

Reply:
True, but the buffer is not only used to drive the display but also to store the pixeldata when drawing before you send it to the display. You need a buffer that can hold tree colors when drawing. I mean, this display is for 3 colors?

If you want the user to handle two buffers themself, I don't see the point in adding `DrawTarget<TriColor>`. 

I'm not sure if we can define the buffer for only b/w in case of `DrawTarget<BinaryColor>` and twice that buffer in case of `DrawTarget<TriColor>`. Can we make `Display2in13bc` generic, i.e. `pub struct Display2in13bc<C: Color>` and use a C::BUFFER_SIZE constant or something?

Reply:
> f you want the user to handle two buffers themself, I don't see the point in adding DrawTarget<TriColor>.
Good point!

> True, but the buffer is not only used to drive the display but also to store the pixeldata when drawing before you send it to the display. You need a buffer that can hold tree colors when drawing. I mean, this display is for 3 colors?

Yeah but you could also redraw the full screen. But I think you can just go forward with your change since it makes much more sense for the general use case.

Reply:
ok, cool. I'll add the `fn` and fix up documentation.